config/render.go

```diff
@@ -0,0 +1,75 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// RenderApp encapsulates the rendered configuration file and the input and
+// output paths.
+type RenderApp struct {
+	renderFlag     string
+	renderedConfig []byte
+}
+
+// EmptyApp creates an empty application
+func EmptyApp() *RenderApp {
+	app := &RenderApp{}
+	return app
+}
+
+// NewRenderApp creates a new Config Rendering application
+func NewRenderApp(configFlag string, renderFlag string) (*RenderApp, error) {
```

Comment:
Starting with easy one, I will change the code to look more like the "-version", i.e. get rid of the "app" function. In that case, I just need to embed more of the logic in app.go.

On the issue of rendering the "config file" vs. "config struct": What I had in mind while writing the code is that after `containerpilot -render` is called you have a config file that can be used as input to `containerpilot -config`. For that to work correctly (I think) I only need to call 'ApplyTemplate' because all the additional work of creating the 'Config struct' will happen once `containerpilot` gets its SIGHUP and 'ParseConfig' gets called. The main idea here is to intentionally not create the struct and instead just render the template and obtain a valid JSON back.

I went through the ParseConfig file again and I did not see the "template filter functions" you talked about. Can you be more specific where they are located?